File: api-tester/src/main/java/com/revenuecat/apitester/java/ProductChangeInfoAPI.java
package com.revenuecat.apitester.java;

import com.revenuecat.purchases.ProductChangeInfo;

@SuppressWarnings({"unused"})
final class ProductChangeInfoAPI {
    static void check(final ProductChangeInfo productChangeInfo) {
        final String oldProductId = productChangeInfo.getOldSku();
        final Integer prorationMode = productChangeInfo.getProrationMode();

        ProductChangeInfo constructedProductChangeInfo = new ProductChangeInfo(
                productChangeInfo.getOldSku(),
                productChangeInfo.getProrationMode()
        );

        ProductChangeInfo constructedProductChangeInfoNullProrationMode =
                new ProductChangeInfo(productChangeInfo.getOldSku(), null);
    }
}

File: api-tester/src/main/java/com/revenuecat/apitester/kotlin/ProductChangeInfoAPI.kt
package com.revenuecat.apitester.kotlin

import com.android.billingclient.api.BillingFlowParams.ProrationMode
import com.revenuecat.purchases.ProductChangeInfo

@Suppress("unused", "UNUSED_VARIABLE", "RemoveExplicitTypeArguments")
private class ProductChangeInfoAPI {
    fun check(productChangeInfo: ProductChangeInfo) {
        with(productChangeInfo) {
            val oldProductId: String = oldSku
            @ProrationMode val prorationMode: Int? = prorationMode

            val constructedProductChangeInfo =
                ProductChangeInfo(
                    oldSku,
                    prorationMode
                )

            val constructedProductChangeInfoSkuOnly = ProductChangeInfo(oldSku)
        }
    }
}
